Mathlib/Data/List/Intervals.lean

```diff
@@ -0,0 +1,261 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+
+! This file was ported from Lean 3 source module data.list.intervals
+! leanprover-community/mathlib commit 7b78d1776212a91ecc94cf601f83bdcc46b04213
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.List.Lattice
+import Mathlib.Data.List.Range
+import Mathlib.Data.Bool.Basic
+/-!
+# Intervals in ℕ
+
+This file defines intervals of naturals. `list.Ico m n` is the list of integers greater than `m`
+and strictly less than `n`.
+
+## TODO
+- Define `Ioo` and `Icc`, state basic lemmas about them.
+- Also do the versions for integers?
+- One could generalise even further, defining 'locally finite partial orders', for which
+  `Set.Ico a b` is `[finite]`, and 'locally finite total orders', for which there is a list model.
+- Once the above is done, get rid of `Data.Int.range` (and maybe `list.range'`?).
+-/
+
+
+open Nat
+
+namespace List
+
+/-- `Ico n m` is the list of natural numbers `n ≤ x < m`.
+(Ico stands for "interval, closed-open".)
+
+See also `data/set/intervals.lean` for `set.Ico`, modelling intervals in general preorders, and
+`multiset.Ico` and `finset.Ico` for `n ≤ x < m` as a multiset or as a finset.
+ -/
+def Ico (n m : ℕ) : List ℕ :=
+  range' n (m - n)
+#align list.Ico List.Ico
+
+namespace Ico
+
+theorem zero_bot (n : ℕ) : Ico 0 n = range n := by rw [Ico, tsub_zero, range_eq_range']
+#align list.Ico.zero_bot List.Ico.zero_bot
+
+@[simp]
+theorem length (n m : ℕ) : length (Ico n m) = m - n :=
+  by
```

Comment:
Please move `by` to the end of the statement. Also elsewhere.
```suggestion
theorem length (n m : ℕ) : length (Ico n m) = m - n := by
  by
```

Reply:
should be fixed now!